Assets/MixedRealityToolkit-SDK/Features/Utilities/Solvers/ControllerFinder.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Definitions.Utilities;
+using Microsoft.MixedReality.Toolkit.Core.EventDatum.Input;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.Devices;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem.Handlers;
+using Microsoft.MixedReality.Toolkit.Core.Managers;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.Utilities.Solvers
+{
+    /// <summary>
+    /// ControllerFinder is a base class providing simple event handling for getting/releasing MotionController Transforms.
+    /// </summary>
+    public abstract class ControllerFinder : MonoBehaviour, IMixedRealitySourceStateHandler
+    {
+        [SerializeField]
+        [Tooltip("The handedness of the controller that should be found.")]
+        private Handedness handedness = Handedness.None;
+
+        /// <summary>
+        /// The handedness of the controller that should be found.
+        /// </summary>
+        public Handedness Handedness
+        {
+            get { return handedness; }
+            set
+            {
+                // We need to refresh which controller we're attached to if we switch handedness.
+                if (handedness != value)
+                {
+                    handedness = value;
+                    RefreshControllerTransform();
+                }
+            }
+        }
+
+        /// <summary>
+        /// The Transform of the currently found controller.
+        /// </summary>
+        protected Transform ControllerTransform;
+
+        private IEnumerable<IMixedRealityManager> BaseDeviceManagers => baseDeviceManagers ?? (baseDeviceManagers = MixedRealityManager.Instance.GetManagers(typeof(IMixedRealityDeviceManager)));
```

Comment:
Not sure this is a good idea.

Reply:
Yikes, missed that.  I cannot think of any good reason why any component outside of the device layer would need to enumerate the managers, especially the devices.
The abstraction is there for a reason to ensure maximum compatibility.

Reply:
Thanks!

Reply:
Easy, use the pose synchronizer

Reply:
>The abstraction is there for a reason to ensure maximum compatibility.

Exactly, that's why I'm only asking for and calling methods contained in the base device interface abstraction. Complete compatibility regardless of the platform I'm on. Other SDK components talk to systems using `GetManager` as well.

What's a scenario where this breaks or does something "wrong"?

Reply:
Exactly the scenario you're proposing.

Reply:
Nothing breaks and nothing wrong happens. Please be more specific instead of restating what you've already said.

Reply:
I'm not sure how I can get more specific as we've clearing pointed out everything wrong in other sections of the comments and issues related to this. I'm going to politely ask you not to work on this anymore, as I'm not entirely confident you understand the implications of the changes you're requesting to make.

Reply:
I'm making one change to the device layer, and the rest are SDK changes to solvers only. Feel free to enumerate the implications in this thread.

Reply:
> I'm not sure how I can get more specific as we've clearing pointed out everything wrong in other sections of the comments and issues related to this. I'm going to politely ask you not to work on this anymore, as I'm not entirely confident you understand the implications of the changes you're requesting to make.

There have been quite a few "this is wrong and vague comments on this PR relating to lack of confidence in various contributor's understandings of the architecture and the intent of these changes, as well as requests for elaboration on architecture.

I do agree that we need clarity (tasks?) for feature additions and I would like to understand why this "is not compatible with the way the MRTK is architected,". Documentation beyond the layer diagrams would go a long way towards helping here.

Reply:
Let's try to come to a common understanding on what the goal is and why things need to be done in certain ways (or not).

My goal is to avoid heated discussions over async text as they can rapidly devolve.